hotsos/core/ycheck/engine/properties/requires/types/snap.py

```diff
@@ -22,19 +22,61 @@ def installed_revisions(self):
         _revisions = []
         for p in self.installed:
             _revisions.append(self.packaging_helper.get_revision(p))
+
         return _revisions
 
-    def package_revision_within_ranges(self, pkg, ranges):
+    @cached_property
+    def installed_channels(self):
+        _channels = []
+        for p in self.installed:
+            _channels.append(self.packaging_helper.get_channel(p))
+
+        return _channels
+
+    def package_info_matches(self, pkg, pkg_info):
+        """
+        If snap package has versions and/or channels we check them here.
+        """
         result = False
-        revision = int(self.packaging_helper.get_revision(pkg))
-        # compare revisions for the time being
-        for item in sorted(ranges, key=lambda i: i['max'],
-                           reverse=True):
-            r_max = int(item['max'])
-            r_min = int(item['min'])
-            if r_min <= revision <= r_max:
-                result = True
-        return result
+
+        # Ranges must be specified as min:max
+        if any((i.get('min') is None and i.get('max') is not None) or
+               (i.get('max') is None and i.get('min') is not None)
+                for i in pkg_info):
+            raise Exception("ranges for package '{}' contain one or more "
+                            "range that do not specify both a min and a max "
+                            "revision".format(pkg))
+
+        _channel_to_check = None
+        if any(i.get('min') is not None for i in pkg_info):
+            revision = int(self.packaging_helper.get_revision(pkg))
+            # compare revisions for the time being
+            for item in sorted(pkg_info, key=lambda i: i['max'],
+                               reverse=True):
+                r_max = int(item['max'])
+                r_min = int(item['min'])
+                if r_min <= revision <= r_max:
+                    if 'channel' in item:
+                        _channel_to_check = item['channel']
+                        break
+
+                    log.debug("match snap revision within range %s:%s", r_min,
+                              r_max)
+                    return True
+
+        if not all(i.get('channel') is None for i in pkg_info):
+            channel = self.packaging_helper.get_channel(pkg)
+            if _channel_to_check is not None:
+                if _channel_to_check != channel:
+                    result = False
+            else:
+                for item in pkg_info:
+                    if item['channel'] == channel:
+                        result = True
+                        log.debug("match snap channel %s", channel)
+                        break
+
+            return result
```

Comment:
Still seems indented too much, should be same level as 'result = False'

Reply:
doh you are absolutely right, fixing now.

Reply:
this was obviously not sufficiently covered by tests so im fixing that too